### PR TITLE
fix(flask): remove deprecated hooks in flask 2.3.0

### PR DESCRIPTION
**Pull request by gnufede**
Fixes Flask patching errors for the Flask release 2.3.0, and adjusts the CI pipeline:

- Deprecated hooks `Flask.before_first_request` and `Blueprint.before_app_first_request`
- Dropped support for Python 3.7

Fixes https://github.com/DataDog/dd-trace-py/issues/5671

## Checklist
- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] PR description includes explicit acknowledgement/acceptance of the performance implications of this PR as reported in the benchmarks PR comment.

## Reviewer Checklist
- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

**Review by brettlangdon — COMMENTED** [review]
you'll need to run `scripts/compile-and-prune-test-requirements` to update the riot requirements compiled hash files

File: releasenotes/notes/fix-flask-2.3.0-9818330e39b287c2.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    Remove Flask hooks deprecated in version 2.3.0
```
Comment (ZStriker19) [inline releasenotes/notes/fix-flask-2.3.0-9818330e39b287c2.yaml:4]: Should we make this a bit more descriptive and specific for customers? Something like
```suggestion
    flask: Remove patching for Flask hooks ``app.before_first_request`` and ``bp.before_app_first_request`` if Flask version >= 2.3.0.
```

File: tests/contrib/flask/test_request.py [hunk 2]
```diff
@@ -163,9 +164,10 @@ def index():
         # Request tags
         assert spans[0].get_tag(http.QUERY_STRING) == "foo=bar&baz=biz"
 
+    @pytest.mark.skipif(flask_version >= (2, 0, 0), reason="Decoding error thrown in url_split")
```
Comment (ZStriker19) [inline tests/contrib/flask/test_request.py:167]: @gnufede  originally had this set to skip for `2,2,0` but I was still getting a decoding error with `2,0,0`.

**Review by juanjux — APPROVED** [review]

**Comment by wildaces215** [comment]
What is the code that replaces `before_first_request` as it has been deprecated?

**Pull request merged**